src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/ContainingSyntaxContext.cs

```diff
@@ -75,7 +74,7 @@ public bool Equals(ContainingSyntaxContext other)
                 && ContainingNamespace == other.ContainingNamespace;
         }
 
-        public override int GetHashCode() => throw new UnreachableException();
+        public override int GetHashCode() => (ContainingNamespace?.GetHashCode() ?? 0) ^ (!ContainingSyntax.IsEmpty ? ContainingSyntax[0].Identifier.Value.GetHashCode() : 0);
```

Comment:
Will we have multiple `ContainingSyntax` elements? If there is more than one, it seems we should loop over them.

Reply:
I'd like to use `System.HashCode` here to produce the hash code, so I just put together a simple hash code. I can look at improving this (possibly making `System.HashCode` usable in our downlevel projects)